riscv/isa_parser.cc

```diff
@@ -260,6 +274,17 @@ isa_parser_t::isa_parser_t(const char* str, const char *priv)
     bad_isa_string(str, ("Zfinx/ZDinx/Zhinx{min} extensions conflict with Base 'F/D/Q/Zfh{min}' extensions"));
   }
 
+  if ((extension_table[EXT_ZCMP] || extension_table[EXT_ZCMT]) &&
+      (extension_table[EXT_ZCD] || (extension_table['C'] && extension_table['D']))) {
```

Comment:
Does it make more sense to do something like

```
  if (extension_table['C'] && extension_table['D'])
    extension_table[EXT_ZCD] = true;
```

In addition to simplifying this check, this might simplify other code later: namely, you won't need to use `require_either_extension` inside of instructions like `c.fsd`.

Same for Zca and Zcf.

Reply:
This way can simplify the check. There is a question here: "Zcd requires Zca". If we also set Zca = true when C is enabled, it have another question, whether rv32imac_zcb is legal? I didn't find the explicit description about whether C imply Zca and  Zcf.

Reply:
C implies Zca, and so C_Zca, C_Zcb, C_Zcd, and RV32C_Zcf should all be legal.

Reply:
If C implies Zca, whether it implies Zcf and Zcd?

Reply:
That is to say whether C_Zcmt(without D) is legal?

Reply:
C implies Zcd only if D is specified.

C implies Zcf only if F is specified and the base ISA is RV32.

C_Zcmt is legal if neither D nor Zcd is specified.

Reply:
OK. Thanks for your patient explanation.  By the way, "require_extension('F') can be deleted for Zcf instructions? that is to say they can be illegal when zcf and zfinx are specified without  F.

Reply:
As long as you don't break backwards compatibility, that sounds OK.